### PR TITLE
[CN-704] Remove side effects from status update functions

### PR DESCRIPTION
**Pull request by mtyazici**
## Description

Status update functions for all CRs were updating all the fields in the status object. Now, update functions only updates explicit fields.

### CODE REVIEW

File: controllers/hazelcast/queue_controller.go [hunk 1]
```diff
@@ -55,24 +56,31 @@ func (r *QueueReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
...
+		return updateDSStatus(ctx, r.Client, q, recoptions.RetryAfter(retryAfterForDataStructures),
+			withDSState(hazelcastv1alpha1.DataStructurePending),
+			withDSMessage(err.Error()),
+			withDSMemberStatuses(ms))
 	}
 
-	requeue, err := updateDSStatus(ctx, r.Client, q, dsPersistingStatus(1*time.Second).withMessage("Persisting the applied multiMap config."))
+	requeue, err := updateDSStatus(ctx, r.Client, q, recoptions.RetryAfter(1*time.Second),
+		withDSState(hazelcastv1alpha1.DataStructurePersisting),
+		withDSMessage("Persisting the applied multiMap config."),
```
Comment (cagric0) [inline controllers/hazelcast/queue_controller.go:67]:
```suggestion
		withDSMessage("Persisting the applied Queue config."),
```

**Review by cagric0 — CHANGES_REQUESTED** [review]
LGTM!

**Review by cagric0 — APPROVED** [review]

**Review by SeriyBg — APPROVED** [review]

File: controllers/hazelcast/queue_controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mtyazici) [reply controllers/hazelcast/queue_controller.go:67]: test-comment

[the remark this answers is not in the document]
Comment (cagric0) [reply controllers/hazelcast/queue_controller.go:67]: requested-comment

**Pull request merged**